### DISCUSSION TITLE
V module as cross platform compatible

### DISCUSSION THREAD

**Discussion by nedimf** [vlang/v#9521] [Ideas]
## Struggle
I have been playing with Rust integration for iOS, and one thing I noticed is that syntax wise it sucks. 
Why? Because Rust uses C types to actually extern methods from library to end user.
Syntax wise it could be this: 
```rust
#[no_mangle]
pub extern "C" fn give_me_letter_a(count: c_ulong) -> *mut c_char {
    let string = "A".repeat(count as usize);
    let cstring = CString::new(string).unwrap();
    cstring.into_raw()
} 
```
## Solution
This could be solved with V. V exports .c files from .v file with ease. Issue is module that are written in V that actually include other modules does not export to .c files, there for .a files can't be created. And this is where issue blocks this idea.

## Why would this be beneficial for V?
Imagine writing V module that runs on iOS, Android and ASM. This will make building cross platform modules reality.

**Comment by nedimf** [comment]
I have written CLI that compiles V module into arm64/x86_64 static library and generates header files for use in iOS project.
[vmob CLI](https://github.com/nedimf/vmob)

**Reply by dumblob** [reply]
Thanks @nedimf - that tool might come handy soon :wink:.

**Reply by nedimf** [reply]
@dumblob let me know if it works for you. There might be issues, because it's not well tested yet

**Reply by dumblob** [reply]
Sure. That'll take some time in my case as there are tougher challenges in my life to tackle first.

**Reply by nedimf** [reply]
All good, I'm on Discord if you need any help
